### ISSUE TITLE
[BUG] MultiModalPredictor ImagePrediction example not working (shopee dataset) and not running on the MNIST dataset

### ISSUE THREAD

**Issue by spartanjoax** [autogluon/autogluon#3539]
**Bug Report Checklist**

- [X] I provided code that demonstrates a minimal reproducible example.  
- [X] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
I have just installed the latest stable version of AutoGluon and have attempted to run the ImagePrediction classification example found in the AutoGluon webpage (https://auto.gluon.ai/stable/tutorials/multimodal/image_prediction/beginner_image_cls.html). When fitting, the model consistenly only achieves 0.2 accuracy during validation and 0.25 on evaluation and loss is always NaN during training. However when running the Colab notebook, it seems to achieve correct accuracies. (Could it be my version of PyTorch and CUDA?)

I thought it might be a bug in the dataset, so I attempted to run the same code with the MNIST dataset. In this case, the fit method does not go further than inferring the number of classes of the dataset. 

**Expected behavior**

Fit method is capable of converging as in the webpage example results.

**To Reproduce**

```python
from autogluon.multimodal import MultiModalPredictor
from autogluon.tabular import TabularDataset, TabularPredictor
import warnings
import pandas as pd
from tensorflow.keras.datasets import mnist
import autokeras as ak
import uuid
warnings.filterwarnings('ignore')

print('============== AutoGluon test on shopee ==============')

from autogluon.multimodal.utils.misc import shopee_dataset
download_dir = './ag_automm_tutorial_imgcls'
train_data, test_data = shopee_dataset(download_dir, is_bytearray=True)

print('============== Fitting model ==============')
model_path = f"./tmp/{uuid.uuid4().hex}-automm_shopee"
predictor = MultiModalPredictor(label="label", path=model_path)
predictor.fit(train_data=train_data, time_limit=60*8)
print('============== Model fitted ==============')
results = predictor.fit_summary(show_plot=True)
print(results)
print('============== Evaluating model ==============')
scores = predictor.evaluate(test_data)
print('Top-1 test acc: %.3f' % scores["accuracy"])

print('============== AutoGluon test on MNIST ==============')

(x_train, y_train), (x_test, y_test) = mnist.load_data()
train_df = pd.DataFrame({'image': x_train.tolist(), 'label': y_train.tolist()})
test_df = pd.DataFrame({'image': x_test.tolist(), 'label': y_test.tolist()})
print(train_df.shape)
train_data = TabularDataset(train_df)
test_data = TabularDataset(test_df)

print('============== Fitting model ==============')
model_path = f"./tmp/{uuid.uuid4().hex}-automm_mnist"
predictor = MultiModalPredictor(label="label", path=model_path)
predictor.fit(train_data=train_data, time_limit=60*8)
print('============== Model fitted ==============')
results = predictor.fit_summary(show_plot=True)
print(results)
print('============== Evaluating model ==============')
scores = predictor.evaluate(test_data)
print('Top-1 test acc: %.3f' % scores["accuracy"])
```

**Screenshots / Logs**

The following is the output I obtain:

**For shopee dataset**
```shell
============== AutoGluon test on shopee ==============
============== Fitting model ==============
AutoGluon infers your prediction problem is: 'multiclass' (because dtype of label-column == int, but few unique label-values observed).
        4 unique label values:  [0, 3, 1, 2]
        If 'multiclass' is not the correct problem_type, please manually specify the problem_type parameter during predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])
Global seed set to 0
AutoMM starts to create your model. ✨

- AutoGluon version is 0.8.2.

- Pytorch version is 1.10.1+cu111.

- Model will be saved to "/home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee".

- Validation metric is "accuracy".

- To track the learning progress, you can open a terminal and launch Tensorboard:

    # Assume you have installed tensorboard
    tensorboard --logdir /home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee

Enjoy your coffee, and let AutoMM do the job ☕☕☕ Learn more at https://auto.gluon.ai

1 GPUs are detected, and 1 GPUs will be used.
   - GPU 0 name: NVIDIA T550 Laptop GPU
   - GPU 0 memory: 3.48GB/4.29GB (Free/Total)
CUDA version is 11.1.

Using 16bit None Automatic Mixed Precision (AMP)
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name              | Type                            | Params
----------------------------------------------------------------------
0 | model             | TimmAutoModelForImagePrediction | 86.7 M
1 | validation_metric | MulticlassAccuracy              | 0
2 | loss_func         | CrossEntropyLoss                | 0
----------------------------------------------------------------------
86.7 M    Trainable params
0         Non-trainable params
86.7 M    Total params
173.495   Total estimated model params size (MB)
Epoch 0:  50%|█████████████████████▌                     | 60/120 [03:05<03:05,  3.09s/it, loss=nan, v_num=Epoch 0, global step 2: 'val_accuracy' reached 0.20000 (best 0.20000), saving model to '/home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee/epoch=0-step=2.ckpt' as top 3
Epoch 0: 100%|██████████████████████████████████████████| 120/120 [06:14<00:00,  3.12s/it, loss=nan, v_num=Epoch 0, global step 5: 'val_accuracy' reached 0.20000 (best 0.20000), saving model to '/home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee/epoch=0-step=5.ckpt' as top 3
Epoch 1:  22%|█████████▎                                 | 26/120 [01:41<06:07,  3.91s/it, loss=nan, v_num=]Time limit reached. Elapsed time is 0:08:02. Signaling Trainer to stop.
Epoch 1:  39%|████████████████▊                          | 47/120 [02:13<03:27,  2.85s/it, loss=nan, v_num=Epoch 1, global step 6: 'val_accuracy' reached 0.20000 (best 0.20000), saving model to '/home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee/epoch=1-step=6.ckpt' as top 3
Epoch 1:  39%|████████████████▊                          | 47/120 [02:16<03:31,  2.90s/it, loss=nan, v_num=]
Start to fuse 3 checkpoints via the greedy soup algorithm.
Predicting DataLoader 0: 100%|████████████████████████████████████████████████| 5/5 [00:26<00:00,  5.26s/it]
Predicting DataLoader 0: 100%|████████████████████████████████████████████████| 5/5 [00:26<00:00,  5.28s/it]
Predicting DataLoader 0: 100%|████████████████████████████████████████████████| 5/5 [00:26<00:00,  5.29s/it]
AutoMM has created your model 🎉🎉🎉

- To load the model, use the code below:
    ```python
    from autogluon.multimodal import MultiModalPredictor
    predictor = MultiModalPredictor.load("/home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee")
    ```

- You can open a terminal and launch Tensorboard to visualize the training log:
    ```shell
    # Assume you have installed tensorboard
    tensorboard --logdir /home/joax/notebooks/tmp/8e54c1cdf0784e88b1fa49548f206baa-automm_shopee
    ```

- If you are not satisfied with the model, try to increase the training time,
adjust the hyperparameters (https://auto.gluon.ai/stable/tutorials/multimodal/advanced_topics/customization.html),
or post issues on GitHub: https://github.com/autogluon/autogluon

============== Model fitted ==============
{'val_accuracy': 0.20000000298023224, 'training_time': 636.445766210556}
============== Evaluating model ==============
Predicting DataLoader 0: 100%|████████████████████████████████████████████████| 3/3 [00:13<00:00,  4.44s/it]
Top-1 test acc: 0.250
```

**For MNIST dataset**
```shell
AutoGluon infers your prediction problem is: 'multiclass' (because dtype of label-column == int, but few unique label-values observed).
	10 unique label values:  [6, 7, 9, 3, 0, 8, 4, 1, 2, 5]
	If 'multiclass' is not the correct problem_type, please manually specify the problem_type parameter during predictor init (You may specify problem_type as one of: ['binary', 'multiclass', 'regression'])
```

**Installed Versions**

<details>

```python
INSTALLED VERSIONS
------------------
date                   : 2023-09-21
time                   : 15:19:15.602618
python                 : 3.8.18.final.0
OS                     : Linux
OS-release             : 5.15.90.1-microsoft-standard-WSL2
Version                : #1 SMP Fri Jan 27 02:56:13 UTC 2023
machine                : x86_64
processor              : x86_64
num_cores              : 16
cpu_ram_mb             : 7794
cuda version           : 12.525.125.05
num_gpus               : 1
gpu_ram_mb             : [3960]
avail_disk_size_mb     : 938811

accelerate             : 0.16.0
autogluon              : 0.8.2
autogluon.common       : 0.8.2
autogluon.core         : 0.8.2
autogluon.features     : 0.8.2
autogluon.multimodal   : 0.8.2
autogluon.tabular      : 0.8.2
autogluon.timeseries   : 0.8.2
boto3                  : 1.28.52
catboost               : 1.2.2
defusedxml             : 0.7.1
evaluate               : 0.3.0
fastai                 : 2.7.12
gluonts                : 0.13.5
grpcio                 : 1.50.0
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.3.2
jsonschema             : 4.17.3
lightgbm               : 3.3.5
matplotlib             : 3.7.3
mlforecast             : 0.7.3
networkx               : 3.1
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.24.4
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
pandas                 : 1.5.3
Pillow                 : 9.5.0
psutil                 : 5.9.5
pydantic               : 1.10.12
PyMuPDF                : None
pytesseract            : 0.3.10
pytorch-lightning      : 1.9.5
pytorch-metric-learning: 1.7.3
ray                    : 2.3.1
requests               : 2.28.2
scikit-image           : 0.19.3
scikit-learn           : 1.2.2
scikit-learn-intelex   : None
scipy                  : 1.10.1
seqeval                : 1.2.2
setuptools             : 60.2.0
skl2onnx               : None
statsforecast          : 1.4.0
statsmodels            : 0.14.0
tabpfn                 : None
tensorboard            : 2.10.1
text-unidecode         : 1.3
timm                   : 0.9.7
torch                  : 1.10.1+cu111
torchmetrics           : 0.11.4
torchvision            : 0.11.2+cu111
tqdm                   : 4.65.2
transformers           : 4.26.1
ujson                  : 5.8.0
vowpalwabbit           : None
xgboost                : 1.7.6
```

</details>

**Comment by zhiqiangdon** [comment]
Can you try the colab environment? https://colab.research.google.com/github/autogluon/autogluon/blob/stable/docs/tutorials/multimodal/image_prediction/beginner_image_cls.ipynb

My guess is that it's related to your environment.

**Comment by tonyhoo** [comment]
Thanks for reporting the issue. Can you try `torch==1.13.1+cu117` instead of `1.10.1+cu111`? You can do that by running following codes:
```
pip install torch==1.13.1+cu117 torchvision==0.14.1+cu117 --extra-index-url https://download.pytorch.org/whl/cu117
```
